core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -1327,6 +1354,33 @@ private static void loadDirectoryMetadataInternal(RpcContext rpcContext, MountTa
     inodePath.traverse();
   }
 
+  private void maybeFlushJournalToAsyncJournalWriter(RpcContext rpcContext) {
+    if (mUseFileSystemMergeJournalContext
+        && rpcContext.getJournalContext() instanceof MetadataSyncMergeJournalContext) {
+      try {
+        rpcContext.getJournalContext().flush();
+      } catch (UnavailableException e) {
+        // This should never happen because rpcContext is a MetadataSyncMergeJournalContext type
+        // and only flush journal asynchronously
+        LOG.error("Flush journal failed. This should never happen.");
```

Comment:
Maybe should throw an alluxio.exception.status.InternalException here

Reply:
i threw an unchecked exception here as this should never happen